scylla/src/policies/speculative_execution.rs

```diff
@@ -195,3 +195,132 @@ where
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    // Important to start tests with paused clock. If starting unpaused, and calling `tokio::time::pause()`, then
+    // things like `sleep` will advance the timer not fully accurately (I have no idea why), causing
+    // few ms added clock advancement at the end of the test.
+    // Starting paused is done with `#[tokio::test(flavor = "current_thread", start_paused = true)]`.
```

Comment:
:wrench: Please mention that clock pausing is only doable with `current_thread` executor.
There's no need to specify it explicitly, as tests are by default run with a `current_thread` executor; explicitness may help here notice this fact, though.

:seedling: Actually, I'm now a bit afraid that our tests may avoid some problematic interleavings by limiting execution to a single thread. Perhaps we should run more sophisticated tests on `multi_thread` runtime to increase likelihood of catching some of them in some runs?

Reply:
Possibly better testing strategy is to utilize time pausing (which makes clocks deterministic) + setting rng seed on Tokio runtime. We can test with multiple seeds to have more interleavings.
For integration tests, this won't be fully deterministic, because of Scylla. We could create a test-util that simulates Scylla, and make driver "connect" to it directly, eliminating the need for IO. Then the tests should become fully deterministic.

Reply:
What you've written is the opposite of my point. I want the tests to be *less deterministic* in order to catch errors that could arise in (nondeterministic in nature) normal operating of the driver. I want to achieve that by enabling multithreaded tokio runtime. Does your suggestion address this problem and my idea, or is it completely parallel? I lack glue between our approaches.

Reply:
Unrelated to this PR, please open a separate issue and continue the discussion there.

Reply:
> 🔧 Please mention that clock pausing is only doable with `current_thread` executor.

ping @Lorak-mmk